Add special case for MOs print all if left empty, 0, negative, etc.

ghutchis opened the pull request:
e.g.,
```
mo print
all
```

Fairly minor modifications, but really useful, e.g. Avogadro.

ghutchis committed:
Add special case for MOs print all if left empty, 0, negative, etc.

greglandrum commented:
I think the idea is a great one, but it looks like one consequence of this implementation is that MOs always end up being printed. I don't think that was the intention.

How about printing all MOs when num_MOs_to_print < 0?

ghutchis commented:
Well, I think that would need to be != 0, but I'll revise that once I'm back from teaching.

ghutchis committed:
Add a special case for "all" keyword for MO print.

Now will only print MOs if something is specified, with "all"
or negative numbers as special cases for "print all MOs"

greglandrum commented:
This looks good to me.
@psavery are you ok with it too?

psavery commented:
Yep! I tried it out as well for the XeF4 example. From what I can tell, looks like it works fine!
